libs/ledger/include/ledger/fees/chargeable.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include <cstdint>
+
+namespace fetch {
+namespace ledger {
+  
+class Chargeable
+{
+public:
+  // Construction / Destruction
+  Chargeable()                 = default;
+  Chargeable(Chargeable const &) = delete;
+  Chargeable(Chargeable &&)      = delete;
+  virtual ~Chargeable()        = default;
+
+  virtual uint64_t CalculateFee() const = 0;
```

Comment:
It is also nice to document these interfaces. See: https://github.com/fetchai/ledger/blob/ce16976495a7e66f3dec2e0a94f76ef73a68e32e/libs/muddle/include/muddle/muddle_endpoint.hpp#L71